### DISCUSSION TITLE
About whether node can be used in tauri

### DISCUSSION THREAD

**Discussion by laterdayi** [tauri-apps/tauri#7037] [Q&A]
I'm a front-end developer. I'm familiar with node, but I'm not familiar with rust, but I also want to use tauri. What should I do。
Can you use the api in node directly, discarding the rust syntax entirely

**Comment by Shotman** [comment]
Tauri doesn't embed any node runtime, your only solution would be to package your node app with pkg or something similar and then use it as a sidecard

**Comment by laterdayi** [comment]
Do you need to use rust syntax in your project?

**Reply by Shotman** [reply]
You don't need to use Rust you can use frontend APIs provided by Tauri https://tauri.app/v1/api/js/
If those APIs aren't enough for your project you might need to use Rust, but if you don't want to you won't have other choice than do what I said before (ie: Package your app and then use it as a sidecar) since Tauri doesn't embed any other backend runtime except Rust

**Comment by laterdayi** [comment]
If you need an operating system api, do you need to use node dependencies?

**Reply by simonhyll** [reply]
Tauri has nothing to do with Node.js. You can "sidecar" node.js but that's literally just running node.js separately from the actual Tauri app. You could make that an Express REST API for example, but that wouldn't actually be to use Node.js with Tauri, it's just running Node.js separate from Tauri and happy coincidence you developed it in a way that it can interact with what's going on in Tauri. Essentially.

Basically, if you want anything native, you're going to have to use the Tauri JS API. And if that's not enough and you need something else, learning Rust is the only way you can do it. In Tauri you're literally running in a webview, a browser window essentially. There is no Node.js, only browser based javascript at your disposal, and native functions communicate over the IPC to run Rust functions.

**Comment by The-Best-Codes** [comment]
Great tutorial about Node.js as a sidecar on the Tauri website:
https://v2.tauri.app/learn/sidecar-nodejs/

**Comment by alexiokay** [comment]
I want to use npx commands on user system, but I dont want them to manually intall node if they dont have it. how can i just simply run npx commands from my tauri app but using the bundled npx? The sidecar idea seems strange to me, I don't want to hardcode commands in the sidecar just simply use npx
